src/Drush/Commands/DrushCommandsGeneratorBase.php

```diff
@@ -0,0 +1,289 @@
+use DrupalCodeGenerator\Asset\RenderableInterface;
+use DrupalCodeGenerator\Asset\Resolver\ReplaceResolver;
+use DrupalCodeGenerator\Helper\Dumper\DryDumper;
+use DrupalCodeGenerator\Helper\Dumper\FileSystemDumper;
+use DrupalCodeGenerator\Helper\QuestionHelper;
+use DrupalCodeGenerator\Helper\Renderer\TwigRenderer;
+use DrupalCodeGenerator\InputOutput\IO;
+use DrupalCodeGenerator\Twig\TwigEnvironment;
+use DrupalCodeGenerator\Utils;
+use Drush\Commands\DrushCommands;
+use Drush\DrupalFinder\DrushDrupalFinder;
+use Drush\Log\Logger;
+use Psr\Container\ContainerInterface as DrushContainer;
+use Symfony\Component\Console\Input\ArrayInput;
+use Symfony\Component\Console\Input\InputDefinition;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Filesystem\Filesystem;
+use Twig\Loader\FilesystemLoader as TemplateLoader;
+
+/**
+ * Combawa generator commands helper.
+ */
+abstract class DrushCommandsGeneratorBase extends DrushCommands {
+
+  const TEMPLATES_PATH = __DIR__ . '/../../../templates';
+
+  /**
+   * The DrupalFinder utility.
+   *
+   * @var \Drush\DrupalFinder\DrushDrupalFinder
+   */
+  protected DrushDrupalFinder $drupalFinder;
+
+  /**
+   * Command constructor.
+   */
+  public function __construct(
+    protected readonly Filesystem $fileSystem,
+    protected readonly TwigRenderer $renderer,
+  ) {
+    parent::__construct();
+    $this->renderer->setLogger(new Logger($this->output()));
+    $this->renderer->registerTemplatePath(static::TEMPLATES_PATH);
+  }
+
+  /**
+   * Command creator.
+   */
+  public static function createEarly(DrushContainer $drush_container): static {
+    return new static(
+      new Filesystem(),
+      new TwigRenderer(new TwigEnvironment(new TemplateLoader())),
+    );
+  }
+
+  /**
+   * DrupalFinder utility getter.
+   *
+   * @return \Drush\DrupalFinder\DrushDrupalFinder
+   *   The DrupalFinder utility.
+   */
+  public function drupalFinder(): DrushDrupalFinder {
+    if (!isset($this->drupalFinder)) {
+      $this->drupalFinder = $this->processManager()->getDrupalFinder();
+    }
+    return $this->drupalFinder;
+  }
+
+  /**
+   * Run the generation process.
+   *
+   * @param array $options
+   *   The options array given to the command.
+   */
+  protected function generate(array $options): int {
+    // Extract generation data from options.
+    $vars = $this->extractOptions($options);
+
+    // Ask questions to complete existing data.
+    $this->interview($vars);
+
+    // Ensure all vars are correct.
+    $this->validateVars($vars);
+    $vars = Utils::processVars($vars);
+
+    // Show collected data.
+    $summary = $this->getVarsSummary($vars);
+    if (!empty($summary)) {
+      $this->io()->newLine(1);
+      $this->io()->title('Settings summary');
+      $output = array_chunk($summary, 1, TRUE);
+      $this->io()->definitionList(...$output);
+    }
+
+    $proceed = $this->io()->confirm('Are you sure you want to proceed?');
```

Comment:
We should consume the options "--yes" "--no" "--quiet" to prompt this only if requested.

Reply:
Drush provides `-y|--yes` and `--no-interaction` and it works.
If you want to introduce new things, we can think about that in a follow up.